Remove references to `EncryptedSharedPreferences` architecture overview (COMMUNITY)

fynngodau opened the pull request:
`EncryptedSharedPreferences` are no longer used since version 1.15.x through https://github.com/corona-warn-app/cwa-app-android/pull/2468#issuecomment-787045765. Therefore, we should remove the reference to it being used from the documentation.

fynngodau committed:
Remove references to `EncryptedSharedPreferences`

marcauberer committed:
Merge branch 'release/2.15.x' into patch-1

chiljamgossow commented on docs/architecture-overview.md:
```diff
@@ -51,7 +51,7 @@ Licensing: [Apache License 2.0](https://github.com/JodaOrg/joda-time/blob/master
 ### Room
-Room is a persistence library that provides an abstraction layer over SQLite. In contrary against the EncryptedSharedPreferences Room is used for storing more complex data.
+Room is a persistence library that provides an abstraction layer over SQLite. In contrary against the SharedPreferences Room is used for storing more complex data.
 
```
Awkward English sentence :)  Maybe we could correct that as well
As opposed to SharedPreferences, Room..... or
Contrary to SharedPreferences, Room...

fynngodau replied:
Thanks, I felt weird about it too :)

mtwalli committed:
Merge branch 'release/2.15.x' into patch-1

mtwalli committed:
Merge branch 'release/2.15.x' into patch-1

mtwalli committed:
Merge branch 'release/2.15.x' into patch-1

davidlindercwa commented:
Please move to 2.16

jurajkusnier committed:
Merge branch 'release/2.15.x' into patch-1

marcauberer committed:
Merge branch 'release/2.15.x' into patch-1

mtwalli committed:
Merge branch 'release/2.16.x' into patch-1

CV113 reviewed (dismissed):
UA approved

mtwalli committed:
Merge branch 'release/2.16.x' into patch-1

fynngodau committed:
Fix grammar in architecture overview

Apply review https://github.com/corona-warn-app/cwa-app-android/pull/4487#discussion_r764177001

mtwalli committed:
Merge branch 'release/2.16.x' into patch-1

mtwalli commented:
Thank you @fynngodau !

mtwalli committed:
Merge branch 'release/2.16.x' into patch-1

chiljamgossow reviewed (approved):
Nice! Thanks
